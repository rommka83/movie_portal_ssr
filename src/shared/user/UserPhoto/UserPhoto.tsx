import React, { FC, HTMLAttributes } from 'react';
import styles from './userphoto.module.css';
import classNames from 'classnames';

interface IProps {
  src?: string;
}

export const UserPhoto: FC<HTMLAttributes<HTMLDivElement> & IProps> = ({ src }) => {
  return src ? (
    <img src={src} alt="" className={classNames(styles.pic)} />
  ) : (
    <div className={classNames(styles.pic, styles.noUser, 'icon-avatar_20__0 ')}></div>
  );
};
