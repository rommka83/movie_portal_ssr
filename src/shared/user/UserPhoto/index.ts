export { UserPhoto as default } from './UserPhoto';
