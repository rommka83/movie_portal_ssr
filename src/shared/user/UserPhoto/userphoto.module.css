.pic {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  object-fit: cover;
  background-color: var(--gray-16);
}
.noUser {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--gray-16);
  font-size: 25px;
}
