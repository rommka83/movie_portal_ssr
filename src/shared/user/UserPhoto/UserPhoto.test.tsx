import renderer from 'react-test-renderer';
import { UserPhoto } from './UserPhoto';

describe('проверка UserPhoto', () => {
  test('отрисовка UserPhoto с фотографией', () => {
    const component = renderer.create(<UserPhoto src='/aaa' />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('отрисовка UserPhoto без фотографии', () => {
    const component = renderer.create(<UserPhoto />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
