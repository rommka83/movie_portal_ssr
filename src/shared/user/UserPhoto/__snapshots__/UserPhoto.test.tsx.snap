// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка UserPhoto отрисовка UserPhoto без фотографии 1`] = `
<div
  className="pic noUser icon-avatar_20__0 "
/>
`;

exports[`проверка UserPhoto отрисовка UserPhoto с фотографией 1`] = `
<img
  alt=""
  className="pic"
  data-nimg="1"
  decoding="async"
  height={50}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Faaa&w=128&q=75"
  srcSet="/_next/image?url=%2Faaa&w=64&q=75 1x, /_next/image?url=%2Faaa&w=128&q=75 2x"
  style={
    Object {
      "color": "transparent",
    }
  }
  width={50}
/>
`;
